src/osp/Resource/Resource.h

```diff
@@ -30,75 +30,27 @@
 namespace osp
 {
 
-//template <class T>
-//struct Resource;
-
-//using Corrade::Containers::LinkedList;
-//using Corrade::Containers::LinkedListItem;
-
-
-//template <class T>
-//class DependRes : public LinkedListItem<DependRes<T>, Resource<T> >
-//{
-
-//public:
-
-//    DependRes() = default;
-//    DependRes(Resource<T>& resource) :
-//            LinkedListItem<DependRes<T>, Resource<T> >()
-//    { resource.insert(this); }
-
-//    void bind(LinkedList<DependRes<T> >& resource)
-//    { resource.insert(this); }
-
-//    void doErase() override
-//    { LinkedListItem<DependRes<T>, Resource<T> >::list()->cut(this); }
-
-//    constexpr T* get_data()
-//    {
-//        return &(LinkedListItem<DependRes<T>, Resource<T> >::list()->m_data);
-//    }
-
-//};
-
-
-//struct AbstractResource
-//{
-
-//    //std::string m_name; // name is kind of useless when there's path
-//    std::string m_path;
-
-//    bool m_loaded;
-
-//    // TODO: * some sort of type identification, or just use typeof.
-//    //         implement when needed.
-//    //       * a way to identify loading strategy: dir, network, zip, etc
-//    //       * dependency to other resources
-
-//};
-
 template <class TYPE_T>
 struct Resource
 {
     Resource(bool loaded)
      : m_data()
      , m_loaded(loaded)
-     , m_refCount(0)
+     , m_refCount()
     { }
     Resource(bool loaded, TYPE_T&& data)
      : m_data(std::move(data))
      , m_loaded(loaded)
-     , m_refCount(0)
+     , m_refCount()
```

Comment:
this default initializes m_refCount to whatever m_refCounts type's default is.

You'll need to remove this line entirely to make the class-body default actually apply.